src/manage_sql.c

```diff
@@ -32518,6 +32658,8 @@ delete_scanner (const char *scanner_id, int ultimate)
     }
 
 #if ENABLE_AGENTS
+   // Delete agent group tasks related to the scanner.
+  delete_agent_group_tasks_by_scanner (scanner);
```

Comment:
I don't think tasks should be deleted if their assigned scanners are deleted.

As it is now, moving the scanner to the trashcan should just ensure tasks, agents and agent groups correctly identify the scanner as being in the trash, e.g. by updating the scanner_location field. (This would also mean change the previous behavior of deleting the agents and agent groups in this case.)
Fully deleting the scanner would be blocked if there are still any tasks using it. If there are no tasks deleting the agents and agent groups together with fully deleting the scanner would be fine.